controllers/vrg_volsync.go

```diff
@@ -425,3 +425,24 @@ func (v *VRGInstance) disownPVCs() error {
 
 	return nil
 }
+
+// cleanupResources this function deleted all PS, PD and VolumeSnapshots from its owner (VRG)
+func (v *VRGInstance) cleanupResources() error {
+	for idx := range v.volSyncPVCs {
+		pvc := &v.volSyncPVCs[idx]
+
+		if err := v.volSyncHandler.DeleteRS(pvc.Name, pvc.Namespace); err != nil {
+			return err
+		}
+
+		if err := v.volSyncHandler.DeleteRD(pvc.Name, pvc.Namespace); err != nil {
+			return err
+		}
```

Comment:
Previously DeleteRS and DeleteRD were not called from anywhere?

Reply:
Yes, they were called [here](https://github.com/RamenDR/ramen/blob/2dd6459bb290dd7e398b047d25c68a6eb2bda8af/controllers/volsync/vshandler.go#L131) and [here](https://github.com/RamenDR/ramen/blob/2dd6459bb290dd7e398b047d25c68a6eb2bda8af/controllers/volsync/vshandler.go#L306)

Reply:
Old calls are in ReconcileRD and ReconcileRS. There is a comment before DeleteRS in function ReconcileRD: 	
// Check if a ReplicationSource is still here (Can happen if transitioning from primary to secondary)
// Before creating a new RD for this PVC, make sure any ReplicationSource for this PVC is cleaned up first
Same thing for DeleteRD in function ReconcileRS. It is happening in no connection to the ownership.

When the VRG is deleted, previously RS, RD and VolumeShapshot were also deleted because of ownership, but now we need to delete explicitly, when no ownership is set (due to different namespaces)

Reply:
Yeah, this is only needed when the VRG is deleted.